src/remotes/github.rs

```diff
@@ -89,18 +89,36 @@ fn retrieve_github_project_pull_requests(
 /// Extract the project name from a Github origin URL
 pub fn get_github_project_name(origin: &str) -> String {
     trace!("Getting project name for: {}", origin);
-    let project_regex = Regex::new(r".*:(.*/\S+)\.git\w*$").unwrap();
-    let captures = project_regex.captures(origin).unwrap();
-    String::from(&captures[1])
+    let project_regex = Regex::new(r"((http[s]?|ssh)://)?(\S+@)?[^:/]+[:/](?P<project>\S+?)(\.git)?$").unwrap();
```

Comment:
While we're here, can we get rid of this `unwrap()` and properly communicate this error higher in the call stack?

Reply:
Hmm, I could see a need to remove the `unwrap()` calls when evaluating the regex, but when creating it I don't think there's a good way to handle such an error; if unwrap fails, it means the regex engine couldn't parse our regex, which means the code is wrong ; a panic might be ok in this case ?

This is something the user should never see. (Whereas when evaluating the regex, having an result might mean the remote's format isn't compatible with what `git-req` can handle.)

Reply:
I'd be happy to work on eliminating the other `unwrap()` calls of this function though. In that case, the prototype should probably be changed to return either a `Result<String, Error>` or an `Option<String>`. I am quite new to Rust ; which of these type would be prefered in this case?

Reply:
You are correct - my mistake. I meant the `unwrap` on the test portion. I think an `Option` is better in this case as it should be on the calling function to decide what's an error and what isn't.